### ISSUE TITLE
cjose_jws_sign SIGSEGV without set algorithm

### ISSUE THREAD

**Issue by veselov** [cisco/cjose#47]
I see `cjose_jws_sign` does a lot of checks on the input, and returns an error, but if there is no `alg` field set in `CJOSE`, then the code `SIGSEGV` on `strcmp`ing the alg with various values...

**Comment by zandbelt** [comment]
I'm not sure I follow: both [`cjose_jws_sign`](https://github.com/cisco/cjose/blob/master/src/jws.c#L649) and [`cjose_jws_verify`](https://github.com/cisco/cjose/blob/master/src/jws.c#L1043) call  `_cjose_jws_validate_hdr` which checks that `alg` is set . A quick test passing in a header with no `alg` set produces the expected:
````
could not sign JWS: invalid argument [file: jws.c, function: _cjose_jws_validate_hdr, line: 77]
````
Can you elaborate?

**Comment by veselov** [comment]
@zandbelt My bad.
Examining `_cjose_jws_validate_hdr()`:
The code calls `json_object_get()` to get json value for the `alg`. If value is `NULL`, error is returned. Then, however, `json_string_value` is called on this json value, and that result is not checked for 0. `json_string_value` will return `NULL` if the json value is not a string. That is what is actually happening in my case. Not, however, because I stuck a different value type into the `alg` property, but because `json_object_get` gets picked up from a different library (json-c), because my program needs that other library for its own purposes.

I don't think there is anything that can be done in cjose for that.

**Comment by zandbelt** [comment]
Please re-open because it is an issue indeed: not just when creating a JWT but also when verifying one and as such it is vulnerable to DoS attacks. I'll submit a PR.

**Comment by zandbelt** [comment]
@linuxwolf: this very much calls for 0.5.1

**Comment by linuxwolf** [comment]
Fixed with PR #48.
